litmus-portal/frontend/src/views/Targets/TargetCopy/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { Typography } from '@material-ui/core';
+import Done from '@material-ui/icons/DoneAllTwoTone';
+import { ButtonOutlined } from 'litmus-ui';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import useStyles from './styles';
+
+const TargetCopy: React.FC = () => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+
+  const [copying, setCopying] = useState(false);
+
+  function copyTextToClipboard(text: string) {
+    if (!navigator.clipboard) {
+      console.error('Oops Could not copy text: ');
+      return;
+    }
+    setCopying(true);
+    navigator.clipboard
+      .writeText(text)
+      .catch((err) => console.error('Async: Could not copy text: ', err));
+
+    setTimeout(() => setCopying(false), 6000);
+  }
```

Comment:
The clipboard API doesn't work on the non-secure origin envs. this is a problem that we faced in Core, I'm unsure if we are expecting running on such envs. as well, if not please resolve this comment, otherwise it may be of some merit to add a fallback here

Reply:
```
  function fallbackCopyTextToClipboard(text: string) {
    // eslint-disable-next-line no-alert
    window.prompt('Copy to clipboard: Ctrl+C, Enter', text);
  }

  function copyTextToClipboard(text: string) {
    if (!navigator.clipboard) {
      fallbackCopyTextToClipboard(text);
      return;
    }
    setCopying(true);
    navigator.clipboard
      .writeText(text)
      .catch((err) => console.error('Async: Could not copy text: ', err));

    setTimeout(() => setCopying(false), 3000);
  }

```

In core this was the solution we used, if a better solution is possible feel free to implement that